### ISSUE TITLE
German translation not fully loaded

### ISSUE THREAD

**Issue by alexhass** [notepad-plus-plus/notepad-plus-plus#9306]
The German translation for Searching in Settings dialog exits in the XML file, but it not used in the interface.

![2020-12-25_184907](https://user-images.githubusercontent.com/4450722/103139939-fe1f9280-46e1-11eb-9edd-5a06e82b43c3.jpg)

```
                <Searching title="Searching">
                    <Item id="6901" name="Don't fill find field in Find dialog with selected word"/>
                    <Item id="6902" name="Use monospaced font in Find dialog (Need to restart Notepad++)"/>
                    <Item id="6903" name="Find dialog remains open after search that outputs to results window"/>
                    <Item id="6904" name="Confirm Replace All in All Opened Documents"/>
                </Searching>


				<Searching title="Suche">
					<Item id="6901" name="Gewähltes Wort nicht ins Suchfenster der Suche"/>
					<Item id="6902" name="Festbreitenschrift im Dialogfeld 'Suchen' verwenden (Neustart von Notepad++ erforderlich)."/>
					<Item id="6903" name="Der Suchdialog bleibt nach der Suche geöffnet, die im Ergebnisfenster ausgegeben wird"/>
					<Item id="6904" name="Bestätigung für 'Alle Ersetzen' in allen offenen Dokumenten"/>
				</Searching>

```

**Comment by chcg** [comment]
@alexhass Which version of N++ do you use? With

```
Notepad++ v7.9.1   (32-bit)
Build time : Nov  2 2020 - 01:03:56
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Home (32-bit) 
OS Version : 2009
OS Build : 19042.685
Current ANSI codepage : 1252
Plugins : mimeTools.dll NppConverter.dll NppExport.dll 
```

I can't reproduce this issue.

**Comment by alexhass** [comment]
7.9.1 x64

**Comment by sasumner** [comment]
Works fine for me as well:

![image](https://user-images.githubusercontent.com/30118311/103153261-59429b00-475d-11eb-9ee5-45afeedbe52c.png)

**Comment by alexhass** [comment]
How is this possible? 😳

**Comment by alexhass** [comment]
Can you try with portable version, please?

**Comment by sasumner** [comment]
@alexhass My screenshot was from the portable version (I only use portable versions).

**Comment by chcg** [comment]
Maybe you could check inside your german.xml at the top if it is the one from N++ 7.9.1. It should be

> last modified: 2020-10-24 by schnurlos (schnurlos@gmail.com)

and
> 	<Native-Langue name="Deutsch" filename="german.xml" version="7.9.1">

**Comment by yasmise** [comment]
Please try setting language to English (other than Deutsch), then set back to Deutsch.
I can reproduce this issue with following steps:
1. Download N++ 7.8.9 x64 portable version, and extract it.
2. Launch N++ and change language to Deutsch, then close N++.
3. Download ver. 7.9.1 x64 portable, then extract and overwrite to old version.
4. Launch N++ and observe this issue.

**Comment by alexhass** [comment]
@Yasmise that is exactly the upgrade path I used. 7.8.9 to 7.9.1. As you have been able to reproduce the bug I think this is all what is needed and developers can now fix the bug.

**Comment by sasumner** [comment]
>...and developers can now fix the bug

I don't see a "bug" here that can be fixed.

If I follow @yasmise 's reproduction instructions, yes, I see `Searching` in English after step 4.
But this is because the `nativeLang.xml` file is still the one generated from the older version where "Searching" did not exist.
If you do a step 5: Drop down the _Localization_ combobox (that currently is set to "Deutsch") and then **_RESELECT_** "Deutsch", then "Searching" will change from English to the german translation.

When a new _Localization_ is chosen, that localization file, e.g. german.xml, is copied to the N++ root folder and renamed to `nativeLang.xml`.  When N++ is restarted, it looks for that nativeLang.xml file.  Because nothing in the repro steps 1-4 affected that file, the one created by the older software was used by the newer software.

Because updating portable versions in the described manner is pretty much a manual process, there is no reasonable way (for the software) to do things differently.

**Comment by alexhass** [comment]
Something is wrong about your assumtions. There is no file copied to npp root. I‘m updating the files in the msi setup only. After this step they are locked and non admins cannot change them. If users change the language this is done without native lang file in root folder. They cannot change this file.

**Comment by alexhass** [comment]
Please reopen the case until the root cause is 100% identified and the bug fixed.

**Comment by sasumner** [comment]
>Something is wrong about your assumtions

Not really.  My "assumptions" were all based upon a portable version with no permission problems involved, because nothing had been mentioned previously in this thread about those kinds of things coming into play.  

However, I will reopen.

**Comment by alexhass** [comment]
You can reproduce this with my msi setup.

**Comment by sasumner** [comment]
>You can reproduce this with my msi setup.

I don't even know what that means.
But that's OK because I have no interest in working on issues regarding "installer" (which I think this is?).
Someone else will have to look into it.

**Comment by alexhass** [comment]
This is not an installer issue. It is npp itself.

https://www.hass.de/content/notepad-msi-package-enterprise-deployment-available

**Comment by sasumner** [comment]
>This is not an installer issue. It is npp itself.

Ah, shows how little I care about such things.  :-)
Probably someone else will help.

**Comment by yasmise** [comment]
I'm not sure, but `nativeLang.xml` is updated by installer, not Notepad++ itself.
Please check [this installer script](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/b5ec511c38446c743ed190e7f91501d2dd831da4/PowerEditor/installer/nsisInclude/mainSectionFuncs.nsh#L135).

If @alexhass want to make your custom msi installer, then it's your job to implement updating process of `nativeLang.xml`, in my humble opinion.

**Comment by alexhass** [comment]
The file is updated. The msi installer was created to get rid of the nsis installer. It is not the fault of the installer that this issue exists

**Comment by yasmise** [comment]
@alexhass
I've tested your msi installers (v7.8.9 and v7.9.1) and confirmed that your installer doesn't update `nativeLang.xml` under `C:\Users\my_account_name\AppData\Roaming\Notepad++`, while official installer does. I believe this is the root cause of this issue.

The behavior how `nativeLang.xml` is generated is already explained by @sasumner [at this comment](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9306#issuecomment-761626482). According to the comment, I think this issue is not a bug.

**Comment by alexhass** [comment]
That is an interresting detail, but a bug in npp than. :-)

The installer runs at system boot time (system install). No user is logged into the system during installation normally. That means appdata folder is not available. 

If you do a manual install like you done that the user may be logged on, but other users on the computer are not. The setup cannot make changes to appdata.

if npp starts it need to check the lang version file and update if it is outdated. This need to be done automatically.

**Comment by sasumner** [comment]
@donho , if he cares to comment, would be the authority on whether or not this is a bug.

**Comment by dodmi** [comment]
In any case, the behavior could be improved.

As @alexhass states, the installer fails his purpose 
- in any deployment scenario where a service account or system account is used to deploy software (this means with any deployment software I know)
- in any deployment scenario where multiple user accounts exist on the same machine (as only the current user profile can be updated)

The installer behaves (or behaved) inconsistent: While I learned here, that the installer replaces the language file (nativelang.xml) with a later version, it does not for other files like stylers.xml or langs.xml. (I know this, because I had the phenomenon, that in my profile new styles weren't available, while they were in a fresh profile, years ago)

I thought, this was intended, that the installer doesn't touch files in a user's profile in case they were modified by the user. But if it touches the nativelang.xml, this  argument doesn't hold.

a) The user is not intended to change these resource files: Notepad++ should use the files from the installation directory and don't copy them over to AppData at all
b) The user is intended to modify these files: the installer should not touch them at all, Notepad++ should ask the user on a first run after an update to replace the files with newer ones [as this can be done for each user, while the installer can't]; probably a NPP setting is needed for deployment scenarios, so that Notepad++ updates the files automatically without user interaction

**Comment by alexhass** [comment]
Asking a user to update the language files sound like - this will confuse the users. This should be silent from my point of view. Just a startup task that runs in the background. I fully agree the files should stay and used from program files and not copied to appdata.

thanks for explaining again with your words.

**Comment by CennoxX** [comment]
You could ask user only, if the specific files were modified (check that by a hash or something like that). This way normal users wouldn't get that questions, power users that change their files wouldn't be confused.

**Comment by donho** [comment]
Oupdated.
